opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/aggregation/dsl/MetricAggregationBuilder.java

```diff
@@ -57,11 +59,16 @@
 public class MetricAggregationBuilder
     extends ExpressionNodeVisitor<Pair<AggregationBuilder, MetricParser>, Object> {
 
-  private final AggregationBuilderHelper<ValuesSourceAggregationBuilder<?>> helper;
+  private final AggregationBuilderHelper<ValuesSourceAggregationBuilder<?>> valuesSourceAggHelper;
+  private final AggregationBuilderHelper<CardinalityAggregationBuilder> cardinalityAggHelper;
```

Comment:
I checked this helper class and found it's stateless. Can we move the generic type on class to the method to reuse a single instance?

```
@RequiredArgsConstructor
public class AggregationBuilderHelper {

  private final ExpressionSerializer serializer;

  public <T> T build(Expression expression, Function<String, T> fieldBuilder,
                 Function<Script, T> scriptBuilder) {
```

Reply:
Yes, it's done. Thanks!